### PR TITLE
Commonize "material-navigation" module.

### PR DESCRIPTION
**Pull request by terrakok**
Make "material-navigation" multiplatform.

Fixes https://youtrack.jetbrains.com/issue/CMP-1365

## Testing
Checked the module manually
image

```kotlin
commonMain.dependencies {
    implementation(compose.runtime)
    implementation(compose.foundation)
    implementation(compose.material)
    implementation("org.jetbrains.androidx.navigation:navigation-compose:...")
    implementation("org.jetbrains.compose.material:material-navigation:...")
}
```

```kotlin
@Composable
internal fun App() = AppTheme {

    val bottomSheetNavigator = rememberBottomSheetNavigator()
    val navController = rememberNavController(bottomSheetNavigator)

    ModalBottomSheetLayout(bottomSheetNavigator) {
        NavHost(navController, startDestination = "start") {
            composable("start") {
                Box(modifier = Modifier.fillMaxSize(), contentAlignment = Alignment.Center) {
                    Column {
                        Text("START")
                        Button(onClick = { navController.navigate("screen_1") }) {
                            Text("NEXT")
                        }
                        Button(onClick = { navController.navigate("bottomSheet") }) {
                            Text("Bottom Sheet")
                        }
                    }
                }
            }
            composable("screen_1") {
                Box(modifier = Modifier.fillMaxSize(), contentAlignment = Alignment.Center) {
                    Column {
                        Text("SCREEN 1")
                        Button(onClick = { navController.popBackStack() }) {
                            Text("EXIT")
                        }
                    }
                }
            }
            bottomSheet("bottomSheet") {
                Box(contentAlignment = Alignment.Center) {
                    Column {
                        Text("Bottom Sheet")
                        Button(onClick = { navController.navigate("bottomSheet") }) {
                            Text("More sheet")
                        }
                        Button(onClick = { navController.popBackStack() }) {
                            Text("EXIT")
                        }
                    }
                }
            }
        }
    }
}
```

### Features - Multiple Platforms
- New multiplatform module "material-navigation" (in beta status)

### CODE REVIEW

File: compose/material/material-navigation/src/commonMain/kotlin/androidx/compose/material/navigation/BottomSheetNavigator.kt [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+     */
+    public class Destination(
+        navigator: BottomSheetNavigator,
+        content: @Composable ColumnScope.(NavBackStackEntry) -> Unit
+    ) : NavDestination, FloatingWindow {
+        internal val content: @Composable ColumnScope.(NavBackStackEntry) -> Unit
+    }
+
+    internal companion object {
+        internal val NAME: String
```
Comment (MatkovIvan) [inline compose/material/material-navigation/src/commonMain/kotlin/androidx/compose/material/navigation/BottomSheetNavigator.kt]: We don't need this in "expect" part and on Android

Comment (terrakok) [reply compose/material/material-navigation/src/commonMain/kotlin/androidx/compose/material/navigation/BottomSheetNavigator.kt]: done. thanks

**Review by MatkovIvan — APPROVED** [review]

**Comment by terrakok** [comment]
So, what will we do with the module? @MatkovIvan @igordmn

**Comment by igordmn** [comment]
lgtm

**Pull request merged**